Client/Frontend/Browser/TabManager.swift

```diff
@@ -159,6 +168,45 @@ class TabManager : NSObject {
 
         return nil
     }
+    
+    weak var appStateDelegate: AppStateDelegate?
+    
+    private func updateAppState() {
+        let state = mainStore.updateState(.TabTray(tabTrayState: TabTrayState(isPrivate: self.isInPrivateMode)))
+        self.appStateDelegate?.appDidUpdateState(state)
+    }
+    
+    func authorisePrivateMode(navigationController: UINavigationController, toRemainInPrivateMode reverseAuthorisationRequirement: Bool = false) -> Success {
+        if self.isInPrivateMode != reverseAuthorisationRequirement {
+            return succeed()
+        }
+        guard let authInfo = KeychainWrapper.authenticationInfo() else {
+            return succeed()
+        }
+        let success = Success()
```

Comment:
Nit: Don't need this success. You can just call succeed() anywhere you need to return a Succes.